packages/salesforcedx-lightning-lsp-common/src/logger.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Connection } from 'vscode-languageserver';
+
+/**
+ * Intercepts global console logging methods and redirects them to the LSP connection.
+ * This allows language server log messages to appear in the client's output panel
+ * (e.g., VS Code's Output view) instead of the server's local console.
+ *
+ * @param connection - The LSP connection to redirect console output to
+ */
+export const interceptConsoleLogger = (connection: Connection): void => {
```

Comment:
make a WI to get rid of this and have the LS use https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#window_logMessage and the clients listen for those and do the right thing with them.